### PR TITLE
Improve elevation bounds for physical shape layers

### PR DESCRIPTION
**Pull request by dnfield**
This updates the bounds calculation for shadows to be more correct rather than just a hard coded value of 20.0.  The value of 20.0 was particularly problematic for larger elevations, which would introduce clipping of the shadow.

### CODE REVIEW

**Comment by dnfield** [comment]
The value was picked by looking at composition in the Flutter Gallery's "Elevation" demo page - in particular, this value works for the 16 and 24pt elevation demos.

File: flow/layers/physical_shape_layer.cc [hunk 1]
```diff
@@ -51,11 +51,11 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
     set_needs_system_composite(true);
 #else
     // Add some margin to the paint bounds to leave space for the shadow.
-    // The margin is hardcoded to an arbitrary maximum for now because Skia
+    // The margin is set to a multiple of elevation_ for now because Skia
     // doesn't provide a way to calculate it.  We fill this whole region
     // and clip children to it so we don't need to join the child paint bounds.
     SkRect bounds(path_.getBounds());
-    bounds.outset(20.0, 20.0);
+    bounds.outset(elevation_ * 1.5, elevation_ * 1.5);
```
Comment (liyuqian) [inline flow/layers/physical_shape_layer.cc]: The following outset might be more accurate according to the physical model used by Flutter and Skia:
```
ex = (kLightRadius + bounds.width() * 0.5) / kLightHeight;
ey = (kLightRadius + bounds.height() * 0.5) / kLightHeight;
bounds.outset(elevation_ * ex, elevation_ * ey);
```
See 
img_20190305_143908
for the illustration.

Comment (liyuqian) [reply flow/layers/physical_shape_layer.cc]: BTW, be careful about `device_pixel_ratio_`. It seems that all `elevation_, kLightRadius, kLightHeight` should be multiplied with `device_pixel_ratio_`. The ratio of `elevation_` and `kLightHeight` cancels out, but you'll still need `kLightRadius * device_pixel_ratio_`.

Comment (dnfield) [reply flow/layers/physical_shape_layer.cc]: I've updated it and tried to incorporate your diagram as a comment.  Does that look right?

File: flow/layers/physical_shape_layer.cc [hunk 2]
```diff
@@ -51,11 +54,29 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
...
+    //             -------------                        (layer)
+    //                   |
+    //                   |                              (elevation)
+    //                   |
+    // ------------------------------------------------ (canvas)
+    //  -----------                                     (extent of shadow)
+    //
+    // E = lx        }           x = (r + w/2)/h
+    //                } =>
+    // r + w/2 = hx  }           E = (l/h)(r + w/2)
```
Comment (liyuqian) [inline flow/layers/physical_shape_layer.cc]: Looks good! It would be better to add a few legends about what `E, x, h, l` mean.

Comment (dnfield) [reply flow/layers/physical_shape_layer.cc]: Done

**Pull request merged**